name: CheckConsoleErrors
on:
  pull_request:
    branches: [main]
    types: [opened, synchronize]
permissions:
  contents: read
jobs:
  CheckConsoleErrors:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.2 https://github.com/actions/checkout/commit/b4ffde65f46336ab88eb53be808477a3936bae11
      - name: Setup Node.js 20.x
        uses: actions/setup-node@e33196f7422957bea03ed53f6fbb155025ffc7b8 # v3.7.0 https://github.com/actions/setup-node/commit/e33196f7422957bea03ed53f6fbb155025ffc7b8
        with:
          node-version: 20.x
      - name: Install Dependencies
        run: yarn
      - name: Run Build
        run: yarn build:release
        env:
          NODE_OPTIONS: --max_old_space_size=4096
      - name: Run Server
        run: |
          python -m http.server 3000 -d ${{ vars.BUILD_DIR }} &
          sleep 5
      - name: Run Console Errors
        id: consoleErrors
        uses: actions/github-script@d7906e4ad0b1822421a7e6a35d5ca353c962f410 # v6.4.1 https://github.com/actions/github-script/commit/d7906e4ad0b1822421a7e6a35d5ca353c962f410
        with:
          result-encoding: string
          script: |
            const { consoleErrors } = require('./tasks/console-errors.js');
            return await consoleErrors();
      - name: Fail if console errors have been found
        if: ${{ steps.consoleErrors.outputs.result }}
        run: exit 1
